ah-rest-useraccount/src/main/java/br/com/arthub/ah_rest_useraccount/api/v1/filter/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,58 @@
+package br.com.arthub.ah_rest_useraccount.api.v1.filter;
+
+import java.io.IOException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import br.com.arthub.ah_rest_useraccount.api.v1.utils.JwtUtils;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+@Component
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+	@Autowired
+	private JwtUtils jwtUtil;
+    @Autowired
+    private UserDetailsService userDetailsService;
+    
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain) throws ServletException, IOException {
+        final String authorizationHeader = request.getHeader("Authorization");
+
+        String username = null;
+        String jwt = null;
+
+        // Verifica se o header Authorization contém um token JWT
+        if (authorizationHeader != null && authorizationHeader.startsWith("Bearer ")) {
+            jwt = authorizationHeader.substring(7);
+            username = jwtUtil.extractUsername(jwt);
+        }
+
+        // Verifica se o usuário está autenticado e se o token é válido
+        if (username != null && SecurityContextHolder.getContext().getAuthentication() == null) {
+            UserDetails userDetails = this.userDetailsService.loadUserByUsername(username);
```

Comment:
Não tem necessidade de verificar a existência do usuário no banco, até porque quem validará esse ponto será o microserviço de autenticação, foque apenas em validar o token e nada mais. Se vier o token, se ele não esitver expirado e se ele for válido (conseguiu extrair o username dele), prossiga com a requisição, caso contrário barre.

Reply:
Ajustado